File: Easy/190. Reverse Bits/official solution (1).go
// 190. Reverse Bits

// Runtime: 4 ms, faster than 25.91% of Go online submissions for Reverse Bits.

// Memory Usage: 2.5 MB, less than 25.00% of Go online submissions for Reverse Bits.


// Bit by Bit
func reverseBits(num uint32) uint32 {
    ret := uint32(0)
    power := uint32(31)
    for num != 0 {
        ret += (num & 1) << power
        num = num >> 1
        power -= 1
    }

    return ret
}

File: Easy/160. Intersection of Two Linked Lists/official solution (1).java
// 160. Intersection of Two Linked Lists

// Runtime: 823 ms, faster than 5.00% of Java online submissions for Intersection of Two Linked Lists.

// Memory Usage: 42.3 MB, less than 5.71% of Java online submissions for Intersection of Two Linked Lists.


/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    // Brute Force
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        ListNode pA = headA;
        while (pA != null) {
            for (ListNode pB = headB; pB != null; pB = pB.next) {
                if (pA == pB) {
                    return pA;
                }
            }

            pA = pA.next;
        }

        return null;
    }
}